fxtest/lifecycle_test.go

```diff
@@ -117,6 +118,108 @@ func TestLifecycle(t *testing.T) {
 	})
 }
 
+func TestEnforceTimeout(t *testing.T) {
+	// These tests directly call Start and Stop
+	// rather than RequireStart and RequireStop
+	// because EnforceTimeout does not apply to those.
+
+	t.Run("StartHookTimeout", func(t *testing.T) {
+		t.Parallel()
+
+		wait := make(chan struct{}, 1)
+		defer close(wait)
```

Comment:
This doesn't need to be buffered. It never gets any values.
Also worth adding a comment that this exists to ensure that OnStart blocks until the end of this test.
```suggestion
		wait := make(chan struct{})
		defer close(wait) // block OnStart until end of test
```